include/internal/constant_time_locl.h

```diff
@@ -213,18 +213,66 @@ static ossl_inline unsigned char constant_time_eq_int_8(int a, int b)
     return constant_time_eq_8((unsigned)(a), (unsigned)(b));
 }
 
+/* Returns the value unmodified, but avoids optimizations. */
+static ossl_inline unsigned int value_barrier(unsigned int a)
+{
+#if !defined(OPENSSL_NO_ASM) && defined(__GNUC__)
+    unsigned int r;
+    __asm__("" : "=r"(r) : "0"(a));
```

Comment:
The gcc manual says:
```If the C code that follows the asm makes no use of any of the output operands, use volatile for the asm statement to prevent the optimizers from discarding the asm statement as unneeded ```

Reply:
That's fine, since we use the output.

Reply:
The point is that the possible range of r cannot be predicted,
even if a is known to be either 0 or 0xFFFFFFFF

Reply:
So even if several of these barriers are folded together, the result is assumed to be
unpredictable.

Reply:
If they are folded, the two sides of a select have a different input:
`value_barrier(mask) & a) | (value_barrier(~mask) & b)`
so these cannot be folded.

Reply:
For some reason I missed the "no" in "no use".